packages/inventory/src/services/inventory.ts

```diff
@@ -60,7 +61,10 @@ export default class InventoryService implements IInventoryService {
    * @param config - the find configuration to use
    * @return A tuple of inventory items and their total count
    */
-  @InjectEntityManager()
+  @InjectEntityManager(
+    (target) =>
+      target.moduleDeclaration?.resources === MODULE_RESOURCE_TYPE.ISOLATED
```

Comment:
self review: I ve added that as a safe guard but I am unsure if we should keep it since the isolated is not yet available. wdyt @olivermrbl

Reply:
@carlos-r-l-rodrigues  – care to pitch in?

I don't have any strong opinion about this.

Reply:
When splitting into modules and using the orchestrator to call them we aren't sharing the db transaction to any module, even if they are shared.
So I'm not sure if I'm missing something, but in theory the first call to a module method would always start a new transaction.

Reply:
This is exactly what ive changed, if you look at the variant creation and inventory orchestration, i pass the transaction to the context in order for the module to use it when it is not isolated. Otherwise this is causing us issues at the moment. It creates new transaction in loop and consume more connections to the db. With enough concurrent requests, the max number of connections is reached everything gets stuck. It reproduce the same issue we ve got with the script phil was working with and was calling the inventory module in a loop without passing the transaction in the context (few weeks ago)

Reply:
To test that, you can run the create products scenario in the staging repo in the two scenarios, one on develop and one on this branch. On develop none of the requests will complete, on this branch they will just go through as usual. When the module is shared and use the same db, it should use the context entity manager passed by the core since the module become a sort of extension part of the core.

Reply:
I figured this is the main change of it after continuing the review. In this case I think it is ok to keep these changes for now, but we'll have to revisit that and find a solution for isolated modules.

Reply:
Yes indeed, isolated with a connection to the same db might create the issue as well. But i prefer to look at this together when we are both there 💪

Reply:
Also, we might change all the places where the modules are called to pass the manager to the context to avoid other issues now that it is safely managed by those changes. But i didn’t want to include them in that pr, if you could discuss that with @olivermrbl